src/uptane/tuf.h

```diff
@@ -0,0 +1,72 @@
+/**
+ * Base data types that are used in The Update Framework (TUF), part of UPTANE.
+ */
+
+#ifndef AKTUALIZR_UPTANE_TUF_H_
+#define AKTUALIZR_UPTANE_TUF_H_
+
+#include <ostream>
+
+namespace Uptane {
+
+/**
+ * The hash of a file or TUF metadata.  File hashes/checksums in TUF include the length of the object, in order to
+ * defeat infinite download attacks.
+ */
+class Hash {
+ public:
+  enum Type { invalid, sha256, sha512 };
+
+  Hash() : type_(invalid) {}
+  Hash(Type type, const std::string &hash, int64_t length);
+
+  /**
+   * Use to avoid downloading indefinitely long (corrupted) files
+   */
+  int64_t length() const { return length_; }
```

Comment:
There is also the possibility of multiple hashes per target/metadata, so you'd have redundant data.